### ISSUE TITLE
[TypeScript] How to assign variables to styles?

### ISSUE THREAD

**Issue by cloakedch** [simonbengtsson/jsPDF-AutoTable#673]
How is it possible to assign variables to styles?

```ts
const weight: string = 'normal';
const size: number = 14;

autoTable(doc, {
  ...
  styles: {
    fontSize: size,
    fontStyle: weight,
  },
});
```

The TypeScript interpreter then throws for `fontStyle`:
`Type 'string' is not assignable to type '"bold" | "normal" | "italic" | "bolditalic"'`.

Similar issue is with `fillColor`, where an array of numbers is not accepted.

I have seen this post on SO: https://stackoverflow.com/questions/37978528/typescript-type-string-is-not-assignable-to-type

However, a cast to `Styles.fontStyle` does not seem to work:
`Cannot access 'Styles.fontStyle' because 'Styles' is a type, but not a namespace`

Any ideas?

EDIT:
As a workaround, the following works:

```ts
const weight: string = 'normal';
const size: number = 14;

autoTable(doc, {
  ...
  styles: {
    fontSize: size,
    fontStyle: weight as "bold" | "normal" | "italic" | "bolditalic", // alternatively cast to 'any'
  },
});
```

I hope there is a better solution :-)

**Comment by cloakedch** [comment]
@simonbengtsson 
Since you tagged this issue as a question and your bot claims it should not be placed here, I'm going to be honest with you:
It's actually not a question. I know that it is a bug in your source. I just wanted to be nice and avoid claiming that it is a bug without giving you the chance to defend it.

**Comment by simonbengtsson** [comment]
You are right! My mistake. I'll certainly accept a PR loosing the type requirement.
